File: examples/java/core/child-contexts/data-processing.java
public class DataProcessingHandler extends DurableHandler<Map<String, Object>, Map> {

    @Override
    public Map handleRequest(Map<String, Object> event, DurableContext ctx) {
        // Named child context
        var result = ctx.runInChildContext("data_processor", Map.class, childCtx -> {
            return childCtx.step("transform", Map.class,
                stepCtx -> transformData((Map) event.get("data")));
        });

        return result;
    }
}

File: examples/python/operations/child-contexts/data-processing.py
@durable_with_child_context
def data_processing(ctx: DurableContext, data: dict) -> dict:
    """Process data in a child context."""
    result = ctx.step(lambda _: transform_data(data), name="transform")
    return result

@durable_execution
def handler(event: dict, context: DurableContext) -> dict:
    # Named child context
    result = context.run_in_child_context(
        data_processing(event["data"]),
        name="data_processor"
    )
    return result
